recipe/meta.yaml

```
@@ -115,34 +50,36 @@ outputs:
         This compiler metapackage is a convenience ONLY for users.
         Do NOT use this package as a build or host dependency in other
         recipes.  Use the Jinja template function compiler('c') instead.
-
-  - name: cxx-compiler
     requirements:
-      host:
-        - {{ compiler('cxx') }}
-        # Depend on the other metapackages to prevent making duplicate symlinks
-        - {{ pin_subpackage('binutils-meta', exact=True) }}  # [unix]
-        - {{ pin_subpackage('c-compiler', exact=True) }}  # [linux]
       run:
-        - {{ compiler('cxx') }}
-        - {{ pin_subpackage('binutils-meta', exact=True) }}  # [unix]
-    script: build-symlinks.sh  # [unix]
+        - {{ compiler('c') }}
+        - binutils             # [linux]
+        - gcc_linux-aarch64    # [linux and aarch64]
+        - gcc_linux-64         # [linux64]
+        - clang_osx-arm64      # [osx and arm64]
+        - clang_osx-64         # [osx and x86_64]
+        - cctools >=949.0.1         # [osx]
+        - ld64 >=530                # [osx]
+        - llvm-openmp               # [osx]
     test:
       commands:
-        - $CXX --help  # [unix]
-{%- for command in commands['cxx-compiler'] %}  # [unix]
-        - {{ command }} --help > /dev/null  # [unix]
-        - echo "Checking {{ command }} resolves to a path containing ${BUILD}"  # [unix]
-  {%- if cxx_compiler == 'gxx' %}  # [unix]
-        - readlink $(which {{ command }}) | grep -q "${BUILD}"  # [unix]
-  {%- elif cxx_compiler == 'clangxx' %}  # [unix]
-        # The prefixed compilers are a symlink on OSX - check they resolve to the same place
-        - "[ \"$(readlink \"$(which {{ command }})\")\" = \"$(readlink \"$(which ${BUILD}-{{ command }})\")\" ]"  # [unix]
-  {%- endif %}  # [unix]
-{%- endfor %}  # [unix]
+        - $CC --help  # [unix]
+        # Linux-specific environment variable checks
+        - echo "C compiler environment variables:" # [linux]
+        - echo $CC # [linux]
```

It might be clearer if it said what it was printing although the `set -x` obscures the effort slightly.

```suggestion
        - echo CC=$CC # [linux]
```

---

I think the line above it is providing enough to track the intentions: ` - echo "C compiler environment variables:" # [linux]`

---

My contention is that if you have several lines of output it is hard to distinguish both

- which line is which (scanning the script whilst also scanning the output becomes tedious after a couple of lines) 
- whether you have successfully run all the commands you expect to.  If the commands' outputs contain their own name, then that's OK.  How do we tell if we missed the output from `gcc-ar` in the mix?  That is it ran without failure but generated no output (or something).

That said, if you're happy then I'm happy.  😄

---

That's fair. I added that and enabled further testing on linux.